modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBObjectParser.java

```diff
@@ -639,8 +639,8 @@ public Expression percentage(JSONObject json, String tag, Number fallback)
             throw new MBFormatException("\"" + tag + "\" percentage from boolean " + obj
                     + " not supported, expected value between 0 and 1");
         } else if (obj instanceof JSONObject) {
-            throw new UnsupportedOperationException(
-                    "\"" + tag + "\"precentage from Function not yet supported");
+            MBFunction function = new MBFunction(this, (JSONObject) obj);
+            return function.numeric();
```

Comment:
See below, the only example I could find of percentage was "hsl(100,50%,50%)" as part of a CSS color definition. 

This method could be removed and replaced with numeric if you would like to reduce code duplication.